OpenRA.Game/Graphics/HardwarePalette.cs

```diff
@@ -28,7 +28,7 @@ public sealed class HardwarePalette : IDisposable
 		public HardwarePalette()
 		{
 			Texture = Game.Renderer.Context.CreateTexture();
-			readOnlyModifiablePalettes = modifiablePalettes.AsReadOnly();
+			readOnlyModifiablePalettes = modifiablePalettes;
```

Comment:
Looks like `readOnlyModifiablePalettes` can be removed completely now, just pass `modifiablePalettes` directly in the one place this is used. Maybe also rename `modifiablePalettes` to `mutablePalettes` in a separate commit?